internal/querynodev2/segments/collection.go

```diff
@@ -82,8 +82,7 @@ func (m *collectionManager) PutOrRef(collectionID int64, schema *schemapb.Collec
 		return
 	}
 
-	collection := NewCollection(collectionID, schema, meta, loadMeta.GetLoadType())
-	collection.AddPartition(loadMeta.GetPartitionIDs()...)
+	collection := NewCollection(collectionID, schema, meta, loadMeta)
```

Comment:
loadMeta is sent to constructor now, partition can be added by constructor directly.